src/js/controller/qoe.js

```diff
@@ -71,6 +71,9 @@ define([
             trackFirstFrame(model);
             trackStalledTime(model);
         });
+        model.mediaController.on(events.JWPLAYER_VISUAL_QUALITY, function(evt){
+            model.set('visualQuality', evt.level);
+        });
```

Comment:
this should be done in the model's video event handler. It's not qoe's responsibility to set model state.